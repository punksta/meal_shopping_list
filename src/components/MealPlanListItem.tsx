import {
  FlatList,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import React from 'react';
import {MealPLan} from '../data';

interface Props {
  item: MealPLan;
  onPress: (item: MealPLan) => void;
  underlayColor?: string;
}

const defaultUnderlayColor = 'rgba(138,138,138,0.73)';

const MealPlanListItem = ({
  item,
  onPress,
  underlayColor = defaultUnderlayColor,
}: Props) => (
  <TouchableHighlight
    underlayColor={underlayColor}
    onPress={onPress.bind(null, item)}
    style={{
      paddingHorizontal: 16,
      paddingVertical: 16,
      borderBottomWidth: StyleSheet.hairlineWidth,
      borderBottomColor: 'rgba(138,138,138,0.73)',
    }}>
    <View>
      <Text
        numberOfLines={2}
        style={{
          fontSize: 21,
          marginBottom: 4,
        }}>
        {item.title}
      </Text>
      <Text
        style={{
          color: 'gray',
          marginBottom: 8,
        }}>
        {item.id}
      </Text>
      <Text
        style={{
          fontSize: 14,
        }}
        numberOfLines={3}>
        {item.description}
      </Text>
    </View>
  </TouchableHighlight>
);

export default React.memo(MealPlanListItem, (p1, p2) => p1.item === p2.item);
