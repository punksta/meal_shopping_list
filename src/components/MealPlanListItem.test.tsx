import renderer from 'react-test-renderer';
import React from 'react';
import MealPlanListItem from './MealPlanListItem';

describe('MealPlanListItem', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <MealPlanListItem
          onPress={() => null}
          item={{
            id: '1',
            title: 'test',
            description: 'description',
            type: 'type',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
