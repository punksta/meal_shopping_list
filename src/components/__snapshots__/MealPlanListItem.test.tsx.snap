// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealPlanListItem should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "rgba(138,138,138,0.73)",
      "borderBottomWidth": 0.5,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={null}
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontSize": 21,
          "marginBottom": 4,
        }
      }
    >
      test
    </Text>
    <Text
      style={
        Object {
          "color": "gray",
          "marginBottom": 8,
        }
      }
    >
      1
    </Text>
    <Text
      numberOfLines={3}
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      description
    </Text>
  </View>
</View>
`;
